bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/binding/generic/converter/AbstractTransformingChannelHandler.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.thing.binding.generic.converter;
```

Comment:
I assume this class is not meant to be publically exposed - should it be moved to an `internal` package?

Reply:
Done. Since all channel types are covered, I don't think that we need the consumer to extend this class.